### PR TITLE
Add support for feature queries with @supports to garden

### PR DESCRIPTION
**Pull request by Mechrophile**
Hey Joel!  This is the change I spoke to you about earlier.

I decided to make it completely separate (i.e. not refactor to make generics) so that it would be completely compatible with whatever other patches people have made in their own forks.

All the tests are passing, and I've been using it for a while not, all looks good.  I know you're eyeing 2.0.0, but this might at least assist in the meantime :D

### CODE REVIEW

File: src/garden/stylesheet.cljc [hunk 1]
```diff
@@ -60,6 +60,10 @@
   (at-rule :media {:media-queries media-queries
                    :rules rules}))
 
+(defn at-supports [feature-queries & rules]
```
Comment (noprompt) [inline src/garden/stylesheet.cljc]: Needs docstring.

File: src/garden/compiler.cljc [hunk 2]
```diff
@@ -54,7 +54,7 @@
   The returned function accepts two arguments: the media
   expression being evaluated and the current media expression context.
   Both arguments are maps. This is used to provide semantics for nested
-  media queries."}
+  media queries.  Also used to support feature queries"}
```
Comment (noprompt) [inline src/garden/compiler.cljc]: Missing period.

File: src/garden/compiler.cljc [hunk 3]
```diff
@@ -41,9 +41,9 @@
    :vendors []
    ;; A set of properties to automatically prefix with `:vendors`.
    :auto-prefix #{}
-   ;; `@media-query` specific configuration.
+   ;; `@media` and '@supports' query configuration.
```
Comment (noprompt) [inline src/garden/compiler.cljc]: Backticks not quotes.

**Comment by noprompt** [comment]
@Mechrophile This looks good; only a few minor tweaks. The only thing missing is an addition to the `README` which describes this feature. I'm happy to patch the `Changelog` once this is merged.

**Comment by Mechrophile** [comment]
good catches!  they're in

**Pull request merged**